src/SignedPetriNets.jl

```diff
@@ -0,0 +1,60 @@
+module SignedPetriNets
+export AbstractSignedPetriNet, SignedPetriNet, OpenSignedPetriNet, OpenSignedPetriNetOb,
+  AbstractLabelledSignedPetriNet, LabelledSignedPetriNet, OpenLabelledSignedPetriNet, OpenLabelledSignedPetriNetOb
+
+using AlgebraicPetri
+using Catlab, Catlab.CategoricalAlgebra
+
+import AlgebraicPetri: Open
+
+@present SchSignedPetriNet <: SchPetriNet begin
+  L::Ob
+  Z2::AttrType
```

Comment:
We should call this `A` and add an involution `neg` to match the notation in the paper. 

```julia
neg \cdot neg == id(A)
sgn::Hom(L, A)
one::Hom(T, A)
```

Reply:
I moved the definitions to be more closely aligned with the paper schema definitions. This is much more flexible because it allows many definitions of the word "sign" but it does make the API a bit more complex/harder to understand. Take a look and see what you think. 

I think the fact that we are implementing this inside of a programming language and not in the abstract world of a category theorist's mind means we can rely on using an attribute type of `Bool` to dictate the laws of negation/Z2/etc. The weird thing is we have a specific definition of `A`, `one`, and `neg` in this case and we kinda just have to hard code that on creation rather than letting the type system decide it for us.

Reply:
Also here is an example that illustrates the decrease in user experience:

Signed Graph definition where sign is a boolean attribute:

https://github.com/AlgebraicJulia/RegNets.jl/blob/0c20870083bd9530368c7116608bdf9118001e1b/test/Core.jl#L12-L19

Signed Graph definition where sign is an object with involution hom as well as the `one` Hom:

https://github.com/AlgebraicJulia/RegNets.jl/blob/93d641da9939cb90b0b41fe3e882e3cb91fb752a/test/Core.jl#L12-L22

Let me know what you think about this